modular_skyrat/master_files/code/modules/client/preferences.dm

```diff
@@ -175,3 +175,7 @@
 	safe_transfer_prefs_to(character, icon_updates, is_antag)
 	qdel(added_tracker)
 
+// Updates the mob's chat color in the global cache
+/datum/preferences/safe_transfer_prefs_to(mob/living/carbon/human/character, icon_updates = TRUE, is_antag = FALSE)
+	. = ..()
+	GLOB.chat_colors_by_mob_name[character.name] = list(character.chat_color, character.chat_color_darkened) // by now the mob has had its prefs applied to it
```

Comment:
This one should be in our modular folder

Reply:
done